src/functions_framework/event_conversion.py

```diff
@@ -147,13 +152,30 @@ def background_event_to_cloudevent(request) -> CloudEvent:
             uid = data["uid"]
             subject = f"users/{uid}"
 
+    # Handle Firebase DB events.
+    if service == _FIREBASE_DB_CE_SERVICE:
+        # The CE source of firebasedatabase cloudevents includes location information
+        # that is inferred from the 'domain' field of legacy events.
+        if "domain" not in event_data:
+            raise EventConversionException(
+                "Invalid FirebaseDB event payload: missing 'domain'"
+            )
+
+        domain = event_data["domain"]
+        location = "us-central1"
+        if domain != "firebaseio.com":
+            location = domain.split(".")[0]
```

Comment:
I don't have a ton of firebase knowledge: what could this be besides `firebaseio.com`? Is this the intended behavior if the domain contains subdomains?

```
>>> domain = 'foo.bar.baz'
>>> domain.split(".")[0]
'foo'
```

Reply:
yes, that is the intended behaviour as described [here](https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/docs/mapping.md#firebase-rtdb-events-tentative)

- The domain property must be present as a string; if it's missing, the conversion should fail.
- If the domain value is firebaseio.com, the location is us-central1.
- Otherwise, the location is the value of domain before the first period. For example, a domain value of europe-west1.firebasedatabase.app would lead to a location value of europe-west1.